axes/handlers/database.py

```diff
@@ -119,58 +119,48 @@ def user_login_failed(
         # 2. database query: Calculate the current maximum failure number from the existing attempts
         failures_since_start = 1 + self.get_failures(request, credentials)
 
-        # 3. database query: Insert or update access records with the new failure data
-        try:
-            attempt = AccessAttempt.objects.get(
+        if not (settings.AXES_ONLY_USER_FAILURES and username is None):
+            # 3. database query: Insert or update access records with the new failure data
+            attempt, created = AccessAttempt.objects.get_or_create(
                 username=username,
                 ip_address=request.axes_ip_address,
                 user_agent=request.axes_user_agent,
+                defaults={"failures_since_start": failures_since_start}
```

Comment:
We could use defaults to define the other arguments as well here such as `get_data`, `post_data`, `http_accept`, `path_info`, and `attempt_time` and avoid the roundtrips to database in case the attempt does not previously exist and is being created. If it exists, then we need to fetch it and update the counters and metadata, of course.

Reply:
addressed